src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java

```diff
@@ -1281,38 +1281,72 @@ public Group createGroup(JSONObject groupJson)
     public Invite createInvite(JSONObject object)
     {
         final String code = object.getString("code");
-
         final User inviter = object.has("inviter") ? this.createFakeUser(object.getJSONObject("inviter"), false) : null;
 
         final JSONObject channelObject = object.getJSONObject("channel");
-
         final ChannelType channelType = ChannelType.fromId(channelObject.getInt("type"));
-        final long channelId = channelObject.getLong("id");
-        final String channelName = channelObject.getString("name");
 
-        final Invite.Channel channel = new InviteImpl.ChannelImpl(channelId, channelName, channelType);
+        final Invite.Type type;
+        final Invite.Guild guild;
+        final Invite.Channel channel;
+        final Invite.Group group;
+
+        if (channelType == ChannelType.GROUP)
+        {
+            type = Invite.Type.GROUP;
+            guild = null;
+            channel = null;
 
-        final JSONObject guildObject = object.getJSONObject("guild");
+            final String groupName = channelObject.optString("name");
+            final long groupId = channelObject.getLong("id");
+            final String groupIconId = channelObject.optString("icon", null);
 
-        final String guildIconId = guildObject.optString("icon", null);
-        final long guildId = guildObject.getLong("id");
-        final String guildName = guildObject.getString("name");
-        final String guildSplashId = guildObject.optString("splash", null);
-        final VerificationLevel guildVerificationLevel = VerificationLevel.fromKey(Helpers.optInt(guildObject, "verification_level", -1));
-        final int presenceCount = Helpers.optInt(object, "approximate_presence_count", -1);
-        final int memberCount = Helpers.optInt(object, "approximate_member_count", -1);
-        final Set<String> guildFeatures;
+            final JSONArray usernameArray = channelObject.optJSONArray("recipients");
+            final List<String> usernames;
+            if (usernameArray == null)
+                usernames = null;
```

Comment:
Use `Collection.emptyList()`

Reply:
I did this to indicate that the invite didn't contain any information about the group's users. This is also documented in `Invite.Group#getUsers()`
> If the users were not included in the invite, this will return `null`.

Reply:
Returning null is an inconvenience for the end user. What is the purpose of knowing whether this information was present or not?

Reply:
What cases are there in which you don't receive `recipients`? What causes it?

Reply:
Discord only sends them if you use the `with_counts=true` query parameter. This is already dosumented in `Invite.Group#getUsers()`.